R/sandwichMethods.R

```diff
@@ -0,0 +1,180 @@
+#' # We can choose Fisher information matrix instead of default observed information matrix.
+#' vcov(Model, "Fisher")
+#' sandwich::bread(Model, type = "Fisher")
+#' all(vcov(Model, "Fisher") * nrow(netherlandsimmigrant) == sandwich::bread(Model, type = "Fisher"))
+#' @importFrom sandwich bread
+#' @method bread singleR
+#' @exportS3Method
+bread.singleR <- function(object,...) {
+  stats::vcov(object, ...) * as.vector(object$df.residual + length(object$coefficients))
+}
+
+#' @title Heteroscedasticity-Consistent Covariance Matrix Estimation for singleR class
+#' @author Piotr Chlebicki, Maciej Beręsewicz
+#' 
+#' @description S3 method for \code{vcovHC} to handle \code{singleR} class objects. 
+#' Works exactly like \code{vcov.default} the only difference being that this method handles vector generalised linear models.
+#' Updating the covariance matrix in variance/standard error estimation for population size estimator can be done via [singleRcapture::redoPopEstimation()]
+#'
+#' @param x a fitted \code{singleR} class object.
+#' @param type a character string specifying the estimation type, same as in \code{sandwich::vcovHC.default}. HC3 is the default value.
+#' @param omega a vector or a function depending on the arguments residuals (i.e. the derivative of log-likelihood with respect to each linear predictor), diaghat (the diagonal of the corresponding hat matrix) and df (the residual degrees of freedom), same as in \code{sandwich::vcovHC.default}.
+#' @param sandwich logical. Should the sandwich estimator be computed? If set to FALSE only the meat matrix is returned. Same as in [sandwich::vcovHC()]
+#' @param ... additional optional arguments passed to the following functions:
+#' \itemize{
+#'   \item estfun -- for empirical estimating functions.
+#'   \item hatvalues -- for diagonal elements of projection matrix.
+#'   \item sandwich -- only if \code{sandwich} argument in function call was set to \code{TRUE}.
+#' }
+#'
+#' @return Variance-covariance matrix estimation corrected for heteroscedasticity of regression errors.
+#' @seealso [sandwich::vcovHC()] [singleRcapture::redoPopEstimation()]
+#' @examples 
+#' set.seed(1)
+#' N <- 10000
+#' gender <- rbinom(N, 1, 0.2)
+#' eta <- -1 + 0.5*gender
+#' counts <- rpois(N, lambda = exp(eta))
+#' df <- data.frame(gender, eta, counts)
+#' df2 <- subset(df, counts > 0)
+#' mod1 <-  estimate_popsize(formula = counts ~ 1 + gender, data = df2, 
+#' model = "ztpoisson", method = "mle", pop.var = "analytic")
+#' require(sandwich)
+#' HC <- sandwich::vcovHC(mod1, type = "HC4")
+#' Fisher <- vcov(mod1, "Fisher") # variance covariance matrix obtained from 
+#' #Fisher (expected) information matrix
+#' HC
+#' Fisher
+#' # usual results
+#' summary(mod1)
+#' # updated results
+#' summary(mod1, cov = HC,
+#' popSizeEst = redoPopEstimation(mod1, cov = HC))
+#' @importFrom sandwich vcovHC
+#' @method vcovHC singleR
+#' @exportS3Method
+vcovHC.singleR <- function(x, 
```

Comment:
Did you use some of the code from `countreg` package or others? If yes I suggest to add some comment / reference to avoid misunderstanding or accusations that we used someones code without citation.

Reply:
Only from `sandwich`, but fair point, I'll correct it later today.

Reply:
Done.